td-uefi-pi/src/hob.rs

```diff
@@ -49,7 +49,10 @@ pub fn check_hob_length(hob: &[u8], hob_length: usize) -> Option<&[u8]> {
     if phit.header.r#type == HOB_TYPE_HANDOFF
         && phit.header.length as usize >= size_of::<HandoffInfoTable>()
     {
-        if hob_length as u64 == phit.efi_end_of_hob_list - hob.as_ptr() as u64 {
+        if u64::MAX - (hob_length as u64) < hob.as_ptr() as u64 {
```

Comment:
Can we use rust checked_sub function to handle this instead of use u64::MAX?

Reply:
Replaced